libr/bin/p/bin_dyldcache.c

```diff
@@ -869,16 +875,18 @@ static RList *create_cache_bins(RBinFile *bf, RBuffer *cache_buf, cache_hdr_t *h
 			R_FREE (lib_name);
 			deps[i]++;
 
-			ut32 j;
-			for (j = extras[i].dependentsStartArrayIndex; depArray[j] != 0xffff; j++) {
-				bool upward = depArray[j] & 0x8000;
-				ut16 dep_index = depArray[j] & 0x7fff;
-				if (!upward) {
-					deps[dep_index]++;
-
-					char *dep_name = get_lib_name (cache_buf, &img[dep_index]);
-					eprintf ("-> %s\n", dep_name);
-					R_FREE (dep_name);
+			if (extras && depArray) {
+				ut32 j;
+				for (j = extras[i].dependentsStartArrayIndex; depArray[j] != 0xffff; j++) {
+					bool upward = depArray[j] & 0x8000;
+					ut16 dep_index = depArray[j] & 0x7fff;
+					if (!upward) {
+						deps[dep_index]++;
+
+						char *dep_name = get_lib_name (cache_buf, &img[dep_index]);
+						eprintf ("-> %s\n", dep_name);
+						R_FREE (dep_name);
```

Comment:
you can use free (dep_name), no need to NULL it later because the scope is clear